common/app/model/Cached.scala

```diff
@@ -29,41 +31,55 @@ object Cached extends implicits.Dates {
 
   private val tenDaysInSeconds = 864000
 
-  def apply(seconds: Int)(result: Result): Result = {
-    if (cacheableStatusCodes.contains(result.header.status)) cacheHeaders(seconds, result) else result
-  }
-
-  def apply(duration: Duration)(result: Result): Result = {
-    apply(duration.toSeconds.toInt)(result)
-  }
+  case class Hash(string: String)
 
-  def apply(page: Page)(result: Result): Result = {
-    val cacheSeconds = page.metadata.cacheTime.cacheSeconds
-    if (cacheableStatusCodes.contains(result.header.status)) cacheHeaders(cacheSeconds, result) else result
-  }
+  sealed trait CacheableResult
+  case class RevalidatableResult(result: Result, hash: Hash) extends CacheableResult
+  case class WithoutRevalidationResult(result: Result) extends CacheableResult
```

Comment:
I would propose `NoRevalidationResult` here to save a few character but this is up to you